projects/fractally/apps/webapp/src/app/video/components/mute-buttons.tsx

```diff
@@ -31,11 +36,16 @@ export const MuteAudioButton = ({ room, isMuted }: MuteButtonProps) => {
 };
 
 export const MuteVideoButton = ({ room, isMuted }: MuteButtonProps) => {
-    const toggleMute = () => {
-        if (isMuted) {
-            room.videoUnmute();
-        } else {
-            room.videoMute();
+    const toggleMute = async () => {
+        try {
+            if (isMuted) {
+                await room.videoUnmute();
+            } else {
+                await room.videoMute();
+            }
+        } catch (e) {
+            console.error(e);
+            alert("fail to toggle video muting functionality: " + e);
```

Comment:
Similar comment for this too. `toggleMute` does not return a Promise. But the syntax does allow you to catch if it fails. Is this just for debugging, or did adding the `async...await` syntax here actually fix something? Should it fix something?

Whether we await these `room` methods or not, they get executed, and we're not using some return value to toggle the audio/video mute buttons' visual status. That happens in reaction to audio/video (un)mute events we get back from the Signalwire as events. So there's nothing to await here (except for any error message).

Again, this is for my curiosity. I sense there's an opportunity for me to learn something here...or maybe clarify my understanding.

Reply:
Yeah I got the overall flow where we can just call functions and receive feedback from SignalWire events.

This one was only for debugging indeed.